### PR TITLE
Update agent for new 2.0 driver

### PR DESCRIPTION
**Pull request by derekwbrown**
Handles new ioctls for separately setting open/closed list size
imports new header file

revamp stats collection to match driver changes; fewer (but more useful) stats

Undo RSC workaround for kitchen tests as should be no longer needed.

### Motivation

Include latest NPM driver

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/windowsdriver20-11f39d5a55f319c0.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    On Windows, NPM driver adds support for Receive Segment Coalescing support.
```
Comment (jtappa) [inline releasenotes/notes/windowsdriver20-11f39d5a55f319c0.yaml:11]:
```suggestion
    On Windows, NPM driver adds support for Receive Segment Coalescing.
```

File: releasenotes/notes/windowsdriver20-11f39d5a55f319c0.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    This works around Windows bug in which, in some situations, system probe
```
Comment (jtappa) [inline releasenotes/notes/windowsdriver20-11f39d5a55f319c0.yaml:12]:
```suggestion
    This works around a Windows bug which in some situations, system probe 
```

**Review by jtappa — CHANGES_REQUESTED** [review]

File: releasenotes/notes/windowsdriver20-11f39d5a55f319c0.yaml (hunk 2 above)
Comment (derekwbrown) [reply releasenotes/notes/windowsdriver20-11f39d5a55f319c0.yaml:12]: @jtappa 
How about
`This works around a Windows bug which in some situations causes system probe to hang on startup`

Comment (jtappa) [reply releasenotes/notes/windowsdriver20-11f39d5a55f319c0.yaml:12]: That works for me!

**Review by djmitche — APPROVED** [review]
👍🏽 release.json

**Review by jtappa — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Review by brycekahle — APPROVED** [review]

**Pull request merged**